datavisapp/create_db.py

```diff
@@ -1,30 +1,39 @@
 import json
 import sqlite3
-from collections import OrderedDict
-
 import click as click
 import pandas as pd
 
+from collections import OrderedDict
+
 
-def make_table(db, table_name, col_types):
-    conn = sqlite3.connect(db)
-    cursor = conn.cursor()
+def make_table(table_name, col_types, db_name=':memory:'):
+    """
+    Create a new table in a database using column names and types supplied.
+    """
     columns = []
     for col_name, col_type in col_types:
         col_str = f'{col_name} {col_type}'
         columns.append(col_str)
     columns = ', '.join(columns)
     query = f'''create table {table_name} ({columns})'''
     click.echo(f'Creating table {table_name} with SQL command:\n{query}', err=True)
-    cursor.execute(query)
-    conn.close()
 
+    conn = sqlite3.connect(db_name)
+
+    with conn:
+        c = conn.cursor()
+        c.execute(query)
+
+
+def append_csv_to_table(table_name, csv_path, db_name=':memory:'):
+    """
+    Append the data from a csv file to a table in the database.
+    """
+    conn = sqlite3.connect(db_name)
 
-def append_csv_to_table(db, table_name, csv_path):
-    conn = sqlite3.connect(db)
-    df = pd.read_csv(csv_path)
-    df.to_sql(table_name, conn, if_exists='append', index=False)
-    conn.close()
+    with conn:
+        df = pd.read_csv(csv_path)
+        df.to_sql(table_name, conn, if_exists='append', index=False)
```

Comment:
Is this function still necessary? It can simply be replaced by

```python
df = pd.read_csv(csv_path)
append_df_to_table(table_name, df, db_name)
```